src/pages/RepoPage/CoverageTab/subroute/FileExplorer/FlagMultiSelect.jsx

```diff
@@ -0,0 +1,85 @@
+import { isUndefined } from 'lodash'
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+
+import { useLocationParams } from 'services/navigation'
+import { useRepoFlagsSelect } from 'services/repo'
+import { useFlags } from 'shared/featureFlags'
+import Icon from 'ui/Icon'
+import MultiSelect from 'ui/MultiSelect'
+
+const defaultQueryParams = {
+  // without this search here, when updating the params it fills it in for some reason :thinking:
+  search: '',
+  flags: [],
+}
+
+function FlagMultiSelect() {
+  const { params, updateParams } = useLocationParams(defaultQueryParams)
+  const [selectedFlags, setSelectedFlags] = useState(params?.flags)
+  const [flagSearch, setFlagSearch] = useState(null)
+
+  const { coverageTabFlagMultiSelect } = useFlags({
+    coverageTabFlagMultiSelect: false,
+  })
+
+  const {
+    data: flagsData,
+    isLoading: flagsIsLoading,
+    hasNextPage: flagsHasNextPage,
+    fetchNextPage: flagsFetchNextPage,
+  } = useRepoFlagsSelect({
+    filters: { term: flagSearch },
+    options: {
+      suspense: false,
+      enabled: !!coverageTabFlagMultiSelect,
+    },
+  })
+
+  if (!coverageTabFlagMultiSelect) {
+    return null
+  }
+
+  const flagNames = new Set(params?.flags)
+
+  if (!isUndefined(flagsData)) {
+    flagsData?.forEach((flag) => flagNames.add(flag?.name))
+  }
+
+  return (
+    <div className="w-1/6">
```

Comment:
Not sure how to get the multi-select to be a decent enough size without this wrapping `div` scratched my head on it for a while. If you've got an idea **please** lmk.

Reply:
I think this is fine, just giving it a defined length